containerd-shim-spin-config.md

```diff
@@ -0,0 +1,235 @@
+# SKIP 00X - Configuring Runtime Options in the `containerd-shim-spin`
+
+Summary: Configuring runtime options in the `containerd-shim-spin`
+
+Owner: Kate Goldenring <kate.goldenring@fermyon.com> 
+
+Impacted Projects:
+
+- [x] spin-operator
+- [ ] `spin kube` plugin
+- [x] runtime-class-manager
+- [x] containerd-shim-spin
+- [ ] Governance
+- [ ] Creates a new project
+
+Created: 30/05/2024
+
+## Overview
+
+The `containerd-shim-spin` SpinKube executor currently lackd the ability to configure runtime flags for Spin applications. This proposal aims to address this by enabling three types of configurations: Spin execution settings, shim execution settings, and shim execution environment variables. These configurations will be set via Custom Resource Definitions (CRDs) in Kubernetes, specifically extending the SpinExecutor and SpinApp CRDs. This approach will allow operators to easily configure both the application and the platform execution environments.
+
+## Background
+
+SpinKube executors expose various options and arguments in order to configure how an application is executed. Take, for example, the following command:
+
+```sh
+spin up --listen "127.0.0.1:3000" --disable-pooling --log-dir /var/logs --runtime-config-file /var/spin/config.toml
```

Comment:
Does spin have a sub set of wasmtime configuration that  it exposes and allows for tweaking?

Initially the idea to configure these in the proposal makes sense to get us started but the list of configuration options for wasmtime is quite large.  I think we may want to open up an issue in wasmtime to be able to set these via a file.  Which would make it easier to allow for users to tweak changing values longer term.

Reply:
https://github.com/bytecodealliance/wasmtime/issues/8784

Reply:
Thank you for following up with that @Mossaka. Spin doesn't expose many Wasmtime flags, so it doesn't have too many options to configure; though, it may wish to similarly support configuration by file rather than env vars/flags